### PR TITLE
Update ClusterRole permissions

### LINKED ISSUE #2658: [Feature] Fix/tighen permissions in main ClusterRole
### Problem Statement

The main ClusterRole has now incorrect and obsolete permissions which do not align with Kubernetes releases for the past long while.

### Solution Description

Adjust the permissions in the cost-analyzer's ClusterRole.

### Alternatives

_No response_

### Additional Context

_No response_

### Troubleshooting

- [X] I have read and followed the [issue guidelines](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/ISSUE_GUIDELINES.md) and this is a feature request only for the Helm chart.
- [X] I have searched other issues in this repository and mine is not recorded.

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: chipzoller <chipzoller@gmail.com>

## What does this PR change?

Updates the ClusterRole permissions to modernize based on recent versions of Kubernetes.

## Does this PR rely on any other PRs?

No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

No impact.

## Links to Issues or tickets this PR addresses or fixes

Closes #2658

### CODE REVIEW

**Comment by jessegoodier** [comment]
Assuming we should test this? Or, at a minimum mention that we tried a --dry-run?

**Comment by jessegoodier** [comment]
Assuming we should test this? Or, at a minimum mention that we tried a --dry-run?

**Comment by chipzoller** [comment]
This isn't going to have any negative impact as it mainly just removes long obsolete permissions. And a dry-run wouldn't expose a problem anyhow, only study of the cost would if it can't read certain resources.

**Comment by jessegoodier** [comment]
> This isn't going to have any negative impact as it mainly just removes long obsolete permissions. And a dry-run wouldn't expose a problem anyhow, only study of the cost would if it can't read certain resources.

Understood.
but dry run does catch yaml issues.

**Review by JesseGoodier — APPROVED** [review]

**Comment by chipzoller** [comment]
>but dry run does catch yaml issues.

Of course, which is why the CI action does a real install (not a dry run) against the last 6 version of K8s 😄

**Comment by dwbrown2** [comment]
As a follow up, can we document what versions of K8s this chart supports? Would just keep in mind that some on-prem users can be found running older versions...

cc @AjayTripathy @jcharcalla who might have more input.

**Comment by chipzoller** [comment]
I'll add a table to the README file for now.

**Comment by jcharcalla** [comment]
We've seen a customer on k8s v1.22 in the not to distant past.

File: cost-analyzer/templates/cost-analyzer-cluster-role-template.yaml [hunk 1]
```diff
@@ -29,13 +29,13 @@ rules:
...
     resources:
       - configmaps
-      - deployments
       - nodes
       - pods
       - events
       - services
       - resourcequotas
       - replicationcontrollers
+      - replicasets
```
Comment (ameijer) [inline cost-analyzer/templates/cost-analyzer-cluster-role-template.yaml:38]: @chipzoller I think replicasets is in apps? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#replicaset-v1-apps

Comment (chipzoller) [reply cost-analyzer/templates/cost-analyzer-cluster-role-template.yaml:38]: Dang it...

**Pull request merged**